java-checks/src/main/java/org/sonar/java/checks/InnerStaticClassesCheck.java

```diff
@@ -69,13 +70,20 @@ public void visitClass(ClassTree tree) {
     scan(tree.members());
     Boolean oneReference = atLeastOneReference.pop();
     outerClasses.pop();
-    if (!symbol.isStatic() && !oneReference && !outerClasses.isEmpty() && isFirstParentStatic(outerClasses)) {
+    if (!symbol.isStatic() && !oneReference && couldBeDeclaredStatic(tree)) {
       String named = symbol.name().isEmpty() ? "named " : "";
       context.addIssue(tree, this, "Make this a " + named + "\"static\" inner class.");
     }
   }
 
-  private static boolean isFirstParentStatic(Deque<Symbol> outerClasses) {
+  private boolean couldBeDeclaredStatic(ClassTree tree) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) The Cyclomatic Complexity of this method "couldBeDeclaredStatic" is 11 which is greater than 10 authorized. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AMethodCyclomaticComplexity)